src/Localization/Localization/src/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -153,13 +153,20 @@ public IStringLocalizer Create(Type resourceSource)
                 throw new ArgumentNullException(nameof(resourceSource));
             }
 
-            var typeInfo = resourceSource.GetTypeInfo();
-
-            var baseName = GetResourcePrefix(typeInfo);
+            // Get without Add to prevent unnecessary lambda allocation
+            if (_localizerCache.TryGetValue(resourceSource.AssemblyQualifiedName!, out var cache))
```

Comment:
Can GetResourcePrefix return the same result for different Types? Basically, I'm wondering if this change will mean that additional localizers will end up in the cache because the key isn't shared between types.

Either way, add a comment to that effect explaining why AssemblyQualifiedName is preferable to baseName.

Reply:
I had looked at it a while ago when I first worked on it. Will check and come back with an answer.

Reply:
> Can GetResourcePrefix return the same result for different Types

I don't think so, since it will have the type name in the result: `return baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, baseNamespace + ".");`